prov/sockets/src/sock_progress.c

```diff
@@ -2344,19 +2344,30 @@ static int sock_pe_progress_rx_ep(struct sock_pe *pe,
 				  struct sock_ep_attr *ep_attr,
 				  struct sock_rx_ctx *rx_ctx)
 {
-	int ret = 0, i, num_fds;
+	int i, num_fds;
 	struct sock_conn *conn;
 	struct sock_conn_map *map;
-	void *ep_contexts[SOCK_EPOLL_WAIT_EVENTS];
 
 	map = &ep_attr->cmap;
 
 	if (!map->used)
 		return 0;
 
-epoll_wait_retry:
-	num_fds = fi_epoll_wait(map->epoll_set, ep_contexts,
-			SOCK_EPOLL_WAIT_EVENTS, 0);
+	/* Need to extend the number of contexts for epoll */
+	if (map->epoll_ctxs_sz < map->used) {
+		uint64_t new_size = map->used * 2;
+		map->epoll_ctxs = realloc(map->epoll_ctxs,
+		                          sizeof(*map->epoll_ctxs) * new_size);
+		if (!map->epoll_ctxs)
+		{
```

Comment:
If realloc fails, the original buffer is left intact.  But we've lost the buffer in this case.  We need to realloc() to a new variable, and update map->epoll_ctxs only on success.  Otherwise, the original buffer should just be left alone.  In that case, we should just continue, as starvation isn't likely anyway.